examples/seq2seq/distillation.py

```diff
@@ -266,10 +266,10 @@ def create_module(args):
 
 def distill_main(args):
     Path(args.output_dir).mkdir(exist_ok=True)
-    if len(os.listdir(args.output_dir)) > 3 and args.do_train:
-        raise ValueError("Output directory ({}) already exists and is not empty.".format(args.output_dir))
+    check_output_dir(args)
 
     model = create_module(args)
+    args.overwrite_output_dir = True
```

Comment:
Do we need this?

Reply:
yup. I removed the workaround for this issue, and so now it'll be clean.

That is: distillation already adds files to this output_dir, so how can you go about testing it to be empty in finetune - there you had a workaround for testing whether the len(dir) < 3, which is not a clean workaround, IMHO, since if you call finetune directly there should be 0 entries in the output_dir and not 3.

So the code above tells finetune not to check the outputdir.

Does it make sense?

Reply:
Oh, and distillation has just tested that it was empty!

Reply:
Yes, but I would add a comment.
The 3 files logic has a bit more reasoning behind it: If you run a script that ooms on the first training step, there will be a few files created before the OOM, and rerunning with smaller batch size should not require any extra work.

Can we keep the 3 files check in `check_output_dir` or will that break everything?

Reply:
It is not needed in 2 other places, e.g. here:
https://github.com/huggingface/transformers/pull/7281/files#diff-06191e50e8823ea67ac39a7beb110176be3d147f10cb027376b951dee583d2f5L156
so we won't be testing the right thing there.

So I could change this abstracted function to take an optional argument that will check `len(dir) <= expected_items`, which by default will be 0. And in this case it'd be 3.

Or I could revert and leave the hardcoded checks in place.

Reply:
> change abstracted function to take an optional argument
Yes

> default=0


I think finetune.py case also needs three, but idc what the default is.

Thanks!

Reply:
done.